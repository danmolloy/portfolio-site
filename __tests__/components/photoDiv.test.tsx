import { act, fireEvent, render, screen, waitFor } from "@testing-library/react";
import PhotoDiv, { PhotoDivProps } from "@/components/photoDiv";

const mockProps: PhotoDivProps = {
  src: "/mock.jpg",
  height: 100,
  width: 100,
  alt: "mockAlt",
  title: "mockTitle",
  setShowImage: jest.fn()
}

describe("<PhotoDiv />", () => {
  beforeEach(() => {
    render(<PhotoDiv {...mockProps} />)
  })
  it("photo-div is in the document", () => {
    const photoDiv = screen.getByTestId("photo-div")
    expect(photoDiv).toBeInTheDocument()
  })
  it("close btn is in the document and calls setShowImage() on click", () => {
    const closeBtn = screen.getByText("Close")
    expect(closeBtn).toBeInTheDocument()
    act(() => {
      fireEvent.click(closeBtn)
    })
    expect(mockProps.setShowImage).toHaveBeenCalled()
  })
  it("img is in the document with expected attrs", async () => {
    await waitFor(() => {
      const img = screen.getByAltText(mockProps.alt)
      expect(img).toHaveAttribute('title', mockProps.title);
      expect(img).toHaveAttribute('width', String(mockProps.width));
      expect(img).toHaveAttribute('height', String(mockProps.height));
    });
  })
  it("title is in the document", () => {
    const title = screen.getByText(mockProps.title)
    expect(title).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const photoDiv = screen.getByTestId("photo-div")
    expect(photoDiv).toMatchSnapshot()
  })
})