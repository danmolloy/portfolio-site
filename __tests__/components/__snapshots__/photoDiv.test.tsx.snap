// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoDiv /> matches snapshot 1`] = `
<div
  class="z-10 fixed flex flex-col items-center bg-zinc-50 border shadow -mt-16 sm:-mt-0 sm:p-4 rounded"
  data-testid="photo-div"
  tabindex="-1"
>
  <button
    class="p-1 rounded self-end m-1 hover:text-zinc-500"
  >
    Close
  </button>
  <div
    class=""
  >
    <img
      alt="mockAlt"
      data-nimg="1"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=%2Fmock.jpg&w=256&q=75"
      srcset="/_next/image?url=%2Fmock.jpg&w=128&q=75 1x, /_next/image?url=%2Fmock.jpg&w=256&q=75 2x"
      style="color: transparent;"
      title="mockTitle"
      width="100"
    />
  </div>
  <p
    class="my-4"
  >
    mockTitle
  </p>
</div>
`;
